src/WebUI/ImageDetails/ImageDetailsStore.ts

```diff
@@ -33,14 +33,20 @@ export class ImageDetailsStore extends StoreBase {
         return false;
     }
 
+    public hasImageDetailsFetchCompleted(): boolean {
+        return this._hasImageDetailsFetched; 
+    }
+
     private _setHasImageDetailsData = (payload: { [key: string]: boolean } | undefined): void => {
         if (payload) {
             this._hasImageDetails = payload;
+            this._hasImageDetailsFetched = true;
```

Comment:
instead of this, my thinking is we should use hasImageDetails() function which will return undefined when the data is not available. when available we can send false or true. where ever you are using this we can check for === undefined to show the special case, false/true for promise done case

Reply:
Done